lib/actions/build.coffee

```diff
@@ -0,0 +1,81 @@
+# Imported here because it's needed for the setup
+# of this action
+dockerUtils = require('../utils/docker')
+
+exports.getBundleInfo = getBundleInfo = (options) ->
+	Promise = require('bluebird')
+	helpers = require('../utils/helpers')
+
+	if options.application?
+		# An application was provided
+		Promise.try(-> helpers.getAppInfo(options.application))
+		.then (app) ->
+			return [app.arch, app.device_type]
+	else if options.arch? and options.deviceType?
+		return Promise.resolve([options.arch, options.deviceType])
+	else
+		# No information, cannot do resolution
+		return
```

Comment:
This function should always return a promise if it's going to sometimes return a promise, that way we can rely on `func().then()` always being valid, rather than failing horribly when this branch is hit, I'd recommend using `Promise.method`:
```coffee
exports.getBundleInfo = getBundleInfo = Promise.method (options) ->